### ISSUE TITLE
Reach stable definitions of submodules

### ISSUE THREAD

**Issue by mjo22** [mjo22/cryojax#393]
Currently we have these submodules:

- `cryojax.simulator`
- `cryojax.data`
- `cryojax.constants`
- `cryojax.io`
- `cryojax.inference`
    - `cryojax.inference.distributions`
- `cryojax.coordinates`
- `cryojax.image`
    - `cryojax.image.operators`
- `cryojax.rotations`
- `cryojax.utils`
- `cryojax.internal`
- `cryojax.experimental`

It would be good to finalize this list, make adjustments if needed (including naming clarity), and specifically define the scope of each. I can write out one-sentence descriptions of the current state of each submodule.

**Comment by mjo22** [comment]
Here's the working descriptions:

- `cryojax.simulator`: Core capability for simulating cryo-EM images. This includes image formation theory and modeling structural ensembles of proteins.
- `cryojax.data`: Tools for manipulating cryo-EM data, currently just comprising STAR file manipulation and tools for simulation single-particle datasets. In the future, tools here could include compatibility with other software and tools for other workflows, such as tomography. This is named after `torch.utils.data`, as the core object is similar to their `Dataset`.
- `cryojax.image`: Pure functions computed on JAX arrays for images + volumes. For example, functions for padding / cropping.
    - `cryojax.image.operators`: This has two parts to it: 1) filters / masks 2) parameterized functions in fourier / real space. The distinction between the two is that filters / masks are treated as buffers computed as `__init__`-time, while the functions (i.e. the `AbstractFourierOperator` and `AbstractRealOperator`) are computed at run-time.
- `cryojax.io`: I/O routines for common cryo-EM data formats
- `cryojax.inference`: The scope of this submodule is currently unclear.
    - `cryojax.inference.distributions`: Statistical distributions from which to simulate images and also compute a corresponding likelihood. Future inclusions in the currently defined scope could be sampling images from poisson distributions.
-  `cryojax.coordinates`: Making coordinate systems in real/fourier space.
- `cryojax.utils`: Utilities aiming to simplify user-experience working with JAX
- `cryojax.internal`: Internal convenience tools used in cryoJAX core development, exposed to public API for working downstream.
- `cryojax.experimental`: Parts of core cryoJAX that are currently untested / unfinished, but are ready enough to be experimented with in the meantime.

**Comment by mjo22** [comment]
I think there's room for some simplifications to be made and some names to be made more clear. We should discuss.

**Comment by mjo22** [comment]
Addressed in #400, but leaving this open.
